core/templates/pages/exploration-player-page/modals/display-hint-modal.component.ts

```diff
@@ -33,12 +34,17 @@ import { PlayerTranscriptService } from '../services/player-transcript.service';
   templateUrl: './display-hint-modal.component.html'
 })
 export class DisplayHintModalComponent {
-  index: number;
-  COMPONENT_NAME_HINT: string;
-  hint: SubtitledHtml;
-  displayedCard: StateCard;
-  recordedVoiceovers: RecordedVoiceovers;
-  hintContentId: string;
+  // These properties are initialized using Angular lifecycle hooks
+  // and we need to do non-null assertion, for more information see
+  // https://github.com/oppia/oppia/wiki/Guide-on-defining-types#ts-7-1
+  index!: number;
+  hint!: SubtitledHtml | null;
+  // Null is allowed here because displayhint() funtion in
+  // Hints-and-solution-manager.service.ts can return null.
+  hintContentId!: string | null;
+  // Null is allowed here because in SubtitledHtml class in the
+  // Subtitled-html.model.ts file the type of contentID is
+  // Defined as string | null.
```

Comment:
THis comment should be above the varaible.

Reply:
Done!